### PR TITLE
Add info about HTTP client events.

### PR DESCRIPTION
**Pull request by ADmad**
Refs cakephp/cakephp#17416

### CODE REVIEW

File: en/core-libraries/httpclient.rst [hunk 1]
```diff
@@ -479,6 +479,60 @@ instead. You can force select a transport adapter using a constructor option::
...
+The HTTP client triggers couple of events before and after sending a request
+which allows you to modify either the request or response or do other tasks like
+caching, logging etc. You can use the global event manager to setup listeners
+for these events.
+
+HttpClient.beforeSend
+---------------------
+
+    // Somewhere before calling one of the HTTP client's send methods.
+    \Cake\Event\EventManager::instance()->on(
```
Comment (LordSimal) [inline en/core-libraries/httpclient.rst]: ~~Aren't events only dispatched on the client object and not on the global event manager?~~
I just learned instance level events will also appear in the global event manager.

Comment (ADmad) [reply en/core-libraries/httpclient.rst]: Hmm.. This makes me think that suggesting `$client->getEventManager()->on()` would be better.

File: en/core-libraries/httpclient.rst [hunk 2]
```diff
@@ -479,6 +479,59 @@ instead. You can force select a transport adapter using a constructor option::
...
+
+The HTTP client triggers couple of events before and after sending a request
+which allows you to modify either the request or response or do other tasks like
+caching, logging etc.
+
+HttpClient.beforeSend
+---------------------
+
+    // Somewhere before calling one of the HTTP client's methods which makes a request
+    $http->getEventManager()->on(
```
Comment (LordSimal) [inline en/core-libraries/httpclient.rst:493]:
```suggestion
    $client->getEventManager()->on(
```

File: en/core-libraries/httpclient.rst [hunk 3]
```diff
@@ -479,6 +479,59 @@ instead. You can force select a transport adapter using a constructor option::
...
+            // You can use $event->setResult($response) to achieve the same.
+            return new \Cake\Http\Client\Response(body: 'something');
+        }
+    );
+
+HttpClient.afterSend
+---------------------
+
+    // Somewhere before calling one of the HTTP client's methods which makes a request
+    $http->getEventManager()->on(
```
Comment (LordSimal) [inline en/core-libraries/httpclient.rst:516]:
```suggestion
    $client->getEventManager()->on(
```

File: en/core-libraries/httpclient.rst (hunk 2 above)
Comment (ADmad) [reply en/core-libraries/httpclient.rst:493]: All code examples above use the var `$http` for the client instance, hence I used the same for consistency.

Comment (markstory) [reply en/core-libraries/httpclient.rst:493]: The example on 480 uses $client. I can take care of normalizing them all though.

**Pull request merged**